modules/planning/tasks/optimizers/piecewise_jerk_speed/piecewise_jerk_speed_nonlinear_optimizer.h

```diff
@@ -39,6 +39,26 @@ class PiecewiseJerkSpeedNonlinearOptimizer : public SpeedOptimizer {
                          SpeedData* const speed_data) override;
 
   PiecewiseJerkTrajectory1d SmoothSpeedLimit(const SpeedLimit& speed_limit);
+
+  std::unordered_map<int, std::string> ipopt_failure_status_ = {
+      {0, "Solve_Succeeded"},
+      {1, "Solved_To_Acceptable_Level"},
+      {2, "Infeasible_Problem_Detected"},
+      {3, "Search_Direction_Becomes_Too_Small"},
+      {4, "Diverging_Iterates"},
+      {5, "User_Requested_Stop"},
+      {6, "Feasible_Point_Found"},
+      {-1, "Maximum_Iterations_Exceeded"},
+      {-2, "Restoration_Failed"},
+      {-3, "Error_In_Step_Computation"},
+      {-10, "Not_Enough_Degrees_Of_Freedom"},
+      {-11, "Invalid_Problem_Definition"},
+      {-12, "Invalid_Option"},
+      {-13, "Invalid_Number_Detected"},
+      {-100, "Unrecoverable_Exception"},
+      {-101, "NonIpopt_Exception_Thrown"},
+      {-102, "Insufficient_Memory"},
+      {-199, "Internal_Error"}};
```

Comment:
Prefer defining enum in a proto. So you get everything like
- compiling time type check
- value to string
- string to value

Reply:
Done